sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/FromQuery.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Azure.DigitalTwins.Core.Queries.QueryBuilders;
+
+namespace Azure.DigitalTwins.Core.QueryBuilder
+{
+    /// <summary>
+    /// Query object that already contains a select clause.
+    /// </summary>
+    public sealed class FromQuery : QueryBase
+    {
+        private WhereQuery _delegationClause;
+        private AdtQueryBuilder _parent;
+        private FromClause _clause;
+
+        internal FromQuery(AdtQueryBuilder parent, WhereQuery upstreamWhereQuery)
+        {
+            _parent = parent;
+            _delegationClause = upstreamWhereQuery;
```

Comment:
Confusing that it is named `upstreamWhereQuery` to users, but internally stored as `_delegationClause`. Why does the name change?